src/include/miopen/hipoc_program_impl.hpp

```diff
@@ -53,7 +53,7 @@ struct HIPOCProgramImpl
     TargetProperties target;
     boost::filesystem::path hsaco_file;
     hipModulePtr module;
-    boost::optional<TmpDir> dir;
+    mutable boost::optional<TmpDir> dir;
```

Comment:
Can you please help me understand why this was required ?

Reply:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/81c0e4f98a548710626b75ef324774a174a80f0e/src/include/miopen/hipoc_program.hpp#L55
In `HIPOCProgram`, `HIPOCProgramImpl` is const object. I can't change `dir` to `boost::none` without `mutable` keyword.

Reply:
The rule of thumb is that a member can be made mutable when is does not change the externally visible state of the object. This is not the case, because change of `dir` affects the validity of the value returned by `GetCodeObjectPathname()`. In other words, after deletion of temporary directory, the pathname of the CO file is no longer valid.